File: backend/src/main/scala/microservices/flight/objects/CabinInventoryPlannerResponse.scala
// CabinInventoryPlannerResponse defines one cabin inventory entry in flight responses.
package com.typesafe.travel.flight.objects

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class CabinInventoryPlannerResponse(
    inventoryId: String,
    cabinClass: String,
    availableSeats: Int,
    unitPrice: String,
    currency: String,
    status: String,
    isBookable: Boolean
)

object CabinInventoryPlannerResponse:
  given sourceEncoder: Encoder[CabinInventoryPlannerResponse] = deriveEncoder
  given sourceDecoder: Decoder[CabinInventoryPlannerResponse] = deriveDecoder

File: frontend/src/microservices/flight/objects/CabinInventoryPlannerResponse.ts
// 本文件定义 flight 模块的 `CabinInventoryPlannerResponse`，作为舱位响应数据并提供 JSON 编解码。

export type CabinInventoryPlannerResponse = {
  inventoryId: string
  cabinClass: string
  availableSeats: number
  unitPrice: string
  currency: string
  status: string
  isBookable: boolean
}

export const cabinInventoryPlannerResponseFromJson = (json: string): CabinInventoryPlannerResponse =>
  JSON.parse(json) as CabinInventoryPlannerResponse

export const cabinInventoryPlannerResponseToJson = (value: CabinInventoryPlannerResponse): string =>
  JSON.stringify(value)

